uspec/config_module_spec.rb

```diff
@@ -1,36 +1,38 @@
-require_relative 'spec_helper'
-require_relative 'example_config'
+# frozen_string_literal: true
 
-spec 'modules extended with ConfigModule will load configurations' do
-  ExampleConfig.foo == 'bar'
+require_relative "spec_helper"
+require_relative "example_config"
+
+spec "modules extended with ConfigModule will load configurations" do
+  ExampleConfig.foo == "bar"
 end
 
 spec 'modules extended with ConfigModule have "namespace" methods' do
   ExampleConfig.methods.include? :namespace
 end
 
-spec 'config modules have [] methods' do
+spec "config modules have [] methods" do
   ExampleConfig.respond_to? :[]
 end
 
-spec 'config modules return correct data using []' do
-  ExampleConfig[:foo] == 'bar'
+spec "config modules return correct data using []" do
+  ExampleConfig[:foo] == "bar"
 end
 
-spec 'when using [], return nil for nonexistant keys' do
-  ExampleConfig[:nonexistant] == nil
+spec "when using [], return nil for nonexistant keys" do
+  ExampleConfig[:nonexistant].nil?
 end
 
-spec 'nested hash values are properly wrapped' do
+spec "nested hash values are properly wrapped" do
   ExampleConfig.dictionary.class == ConfigModule::ConfigOption
 end
 
-spec 'subkeys are accessible with methods' do
-  ExampleConfig.dictionary.configuration == 'An arrangement of elements in a particular form, figure, or combination.'
+spec "subkeys are accessible with methods" do
+  ExampleConfig.dictionary.configuration == "An arrangement of elements in a particular form, figure, or combination."
 end
 
-spec 'subkeys work with .each' do
-  text = String.new
+spec "subkeys work with .each" do
+  text = "".dup
```

Comment:
What is happening with the `dup`?

Reply:
It's checking against Ruby 2.0, so it's saying to not use the `String.new`. However, if I don't use `.dup` then the string will be frozen in Ruby 2.3+, and the `<<` later will fail. I can just turn it back into a String.new, as this isn't a blocking failure.

Reply:
I used another method that it won't complain about.

Reply:
This whole bit could be replaced with an inject or something cleaner later. Either way is fine for now, we'll create an issue to address it later.